packaging/nuget/nservicebus.core.nuspec

```diff
@@ -18,7 +18,9 @@
       <frameworkAssembly assemblyName="System" targetFramework="net452" />
       <frameworkAssembly assemblyName="System.Core" targetFramework="net452" />
     </frameworkAssemblies>
-    <dependencies/>
+    <dependencies>
+      <dependency id="System.ValueTuple" version="4.3.0" />
```

Comment:
Thinking through this, here's what I've come up with:

- Adding the dependency wouldn't be a compile break, because if you pull the new package down, you'll get the `System.ValueTuple` as part of that.
- Because we'd only be referencing the tuple stuff in the new learning transport code, if someone cracked open the nuget package and pulled out the binary to replace it in an existing project, this wouldn't be a binary break, because the already-compiled project wouldn't be using the learning transport, so the code that needs it would never execute.
- However, if any other code was updated to use tuples, then that could cause a binary break 😢 
- We could look into ILMerging it instead to sidestep the issue, though maybe the current licensing doesn't allow for that. I do know MS is planning on making sure their open source nuget packages are all licensed the same as the repos they are built from, but not sure that's happened yet.

So, in general, I'd like to be able to take advantage of all C# 7 features, but I can see it not being worth doing now vs. waiting for 7.0.